File: Ex-20/ex_20_java/Universidade.java
import java.util.LinkedList;
import java.util.List;

public class Universidade {
    private String nome;
    private List<Departamento> lDepartamentos;
    private int nDepartamentos;
    private static final int MAX_DEPARTAMENTOS = 50;

    // Construtora & Destrutora
    public Universidade() {
        this.nome = "";
        this.lDepartamentos = new LinkedList<Departamento>();
        this.nDepartamentos = 0;
    }

    // Métodos
    public String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public void addDepartamento(Departamento departamento) {
        if (this.nDepartamentos < MAX_DEPARTAMENTOS) {
            this.lDepartamentos.add(0, departamento);
            this.nDepartamentos++;
        } else {
            System.err.println("Máximo 50 departamentos");
            System.exit(1);
        }
    }

    public void imprimeDepartamentos() {
        System.out.println("Departamentos: ");
        for (Departamento departamento : this.lDepartamentos) {
            System.out.println("\t -> " + departamento.getNome());
        }
    }
}

File: Ex-21/ex_21_python/Universidade.py
import sys


class Universidade:
    MAX_DEPARTAMENTOS = 50

    # Construtora & Destrutora
    def __init__(self):
        self.nome = ""
        self.lDepartamentos = []
        self.nDepartamentos = 0

    # Métodos
    def getNome(self):
        return self.nome

    def setNome(self, nome):
        self.nome = nome

    def addDepartamento(self, departamento):
        if self.nDepartamentos < self.MAX_DEPARTAMENTOS:
            self.lDepartamentos.insert(0, departamento)
            self.nDepartamentos += 1
        else:
            print("Máximo 50 departamentos", file=sys.stderr)
            sys.exit(1)

    def imprimeDepartamentos(self):
        print("Departamentos: ")
        for departamento in self.lDepartamentos:
            print("\t -> " + departamento.getNome())
